frontend/public/components/environment.jsx

```diff
@@ -266,36 +272,57 @@ export const EnvironmentPage = connect(stateToProps)(
     }
 
     render() {
-      const {errorMessage, success, inProgress, currentEnvVars, stale, configMaps, secrets} = this.state;
+      const {errorMessage, success, inProgress, currentEnvVars, stale, configMaps, secrets, containerKey} = this.state;
       const {rawEnvData, readOnly, obj} = this.props;
       const isContainerArray = _.isArray(rawEnvData);
+      const containerDropdown = <ContainerDropdown
+        currentKey={containerKey}
+        containers={rawEnvData}
+        onChange={this.selectContainer} />;
 
       if (!configMaps || !currentEnvVars || !secrets) {
         return <LoadingBox />;
       }
-      const containerVars = currentEnvVars.map((envVar, i) => {
-        const keyString = isContainerArray ? rawEnvData[i].name : obj.metadata.name;
-        return <React.Fragment key={keyString}>
+      const envVar = currentEnvVars[containerKey];
+      const owners = _.get(obj.metadata, 'ownerReferences', [])
+        .map((o, i) => <ResourceLink key={i} kind={referenceForOwnerRef(o)} name={o.name} namespace={obj.metadata.namespace} title={o.uid} />);
+      const resourceName = _.get(obj.metadata, 'name', '');
+      const containerVars =
+        <React.Fragment>
+          { readOnly &&
```

Comment:
It doesn't look like this is always getting set correctly. Replication controllers owned be a deployment config should have a read only form. Same for replica sets owned by a deployment.

Reply:
Hmm Yeah we need to discuss this. I wasn't aware of these two read-only cases. My impression was pods and builds were the only read only entities for env vars. So are you saying that the pattern is all pods, all builds, and all rc's ownedby a deployment config and replica sets owned by deployments? Does that cover the read only use case in its entirety?